src/libraries/System.Net.Mail/src/System/Net/Mail/MailAddressCollection.cs

```diff
@@ -51,25 +51,26 @@ internal void ParseValue(string addresses)
 
         internal string Encode(int charsConsumed, bool allowUnicode)
         {
-            StringBuilder? encodedAddresses = null;
+            var encodedAddresses = new ValueStringBuilder(stackalloc char[256]);
 
             //encode each address individually (except the first), fold and separate with a comma
             foreach (MailAddress address in this)
             {
                 if (encodedAddresses is null)
                 {
                     //no need to append a comma to the first one because it may be the only one.
```

Comment:
```suggestion
            bool first = true;

            //encode each address individually (except the first), fold and separate with a comma
            foreach (MailAddress address in this)
            {
                if (first)
                {
                    first = false;
                    //no need to append a comma to the first one because it may be the only one.
```

Reply:
Yeah, sorry, pushed changes just to have it without finishing it. It would be better to utilize `Length` actually instead of that additional variable or/and even unroll the loop into `if` containing `while` to have no branching inside, but it might be an overkill in that case. What is you opinion?

Reply:
Right, that would work as well